packages/npm-resolver/src/fetch.ts

```diff
@@ -36,7 +42,13 @@ export default async function fromRegistry (
   const uri = toUri(pkgName, registry)
   const response = await fetch(uri, { auth }) as RegistryResponse
   if (response.status > 400) {
+    let info = ''
+    const matched = pkgName.match(semvarRegex)
+    if (matched) {
+      info = ` Did you mean ${matched[1]}?`
+    }
```

Comment:
I think this can be moved to the error class constructor. Then there's no need in a new `info` field